Trackovid19-web/src/app/screens/main/main.component.ts

```diff
@@ -29,36 +29,31 @@ export class MainComponent implements OnInit, OnDestroy {
     private userService: UserService,
     private conditionService: ConfinementStateService,
     private profileService: ProfileServiceService,
-    private route: ActivatedRoute,
     private shareStateService: VideoStateService,
     private renderer: Renderer2,
   ) {
     this.toggleShareCallback = this.toggleShare.bind(this);
 
-    let gdpr = localStorage.getItem('gdpr');
-    gdpr = gdpr !== null ? JSON.parse(gdpr) : false;
-    if (!gdpr) {
-      this.router.navigate(['privacy-terms']);
-    }
-  }
-
-  ngOnInit(): void {
-    this.router.events.forEach(event => {
-      if (
-        event instanceof NavigationEnd &&
-        (event.url.indexOf('/dashboard/status') !== -1 ||
-          event.urlAfterRedirects.indexOf('/dashboard/status') !== -1)
-      ) {
+    router.events.forEach(event => {
+      if (event instanceof NavigationEnd && event.url.indexOf('/dashboard/status') !== -1) {
         const shareVal = localStorage.getItem('share');
-        if (!shareVal || shareVal !== 'true') {
+        if (shareVal && shareVal === 'true') {
           this.showShare = true;
           this.renderer.setStyle(document.body, 'overflow', 'hidden');
           this.renderer.setStyle(document.body, 'position', 'fixed');
-          localStorage.setItem('share', 'true');
+          localStorage.setItem('share', 'false');
         }
       }
     });
 
+    let gdpr = localStorage.getItem('gdpr');
```

Comment:
Shouldn't this continue inside ngOnInit?

Reply:
It wasn't inside `ngOnInit`

Reply:
https://github.com/Tech4covid19/trackovid19-web/blob/develop/Trackovid19-web/src/app/screens/main/main.component.ts